massa-module-cache/src/config.rs

```diff
@@ -0,0 +1,19 @@
+use massa_sc_runtime::GasCosts;
+use std::path::PathBuf;
+
+pub struct ModuleCacheConfig {
+    /// Path to the hard drive cache storage
+    pub hd_cache_path: PathBuf,
+    /// Gas costs used to:
+    /// * setup `massa-sc-runtime` metering on compilation
+    /// * debit compilation costs
+    pub gas_costs: GasCosts,
+    /// Default gas for compilation
+    pub compilation_gas: u64,
+    /// Maximum number of entries we want to keep in the LRU cache
+    pub lru_cache_size: u32,
+    /// Maximum number of entries we want to keep in the HD cache
```

Comment:
> Having an entry limitation rather than a Gb limitation makes the hd cache code significantly more convoluted (especially the pruning function) and I don't find its benefits to make up for that.

Ok, that's what I thought happened.

> Also the Gb limitation can be deduced from the entry count because we define a bytecode maximum size. We could have the config parameter unit to be Gb (or Mb) and compute the number of hd cache entries in a worst case scenario.

Yeah we could do something like that. Personnaly I don't think this workaround is wanted (if most modules are small, it seems weird to say "max cache of 10Gb" while using <1Gb for example, even though it is "to specs").

> Compilation costs for cached modules will be associated to the ABIs that produce a bytecode update. Since the gas cost is payed up front, we do not know the CPU cost of compilation yet, so we would have to calibrate a gas cost for those ABIs that take into consideration all of these issues. This has not been done yet, for the time being we need to agree on an approximation, later it will become necessary to define these costs properly. It was considered to make the user pay post compilation but the solutions make storing a bytecode on the blockchain too complicated for users.

Thank you for the explanation!